### PR TITLE
Волхов Михаил 2538

### PR DESCRIPTION
**Pull request by volhovm**
What do I have: Design, Swipe View, Smooth scrolling, Animations, 
More Design, Deletion\Insertion of locations + geolocation, Lots Of Design!~
Not obvious: forecast items (those with red circles) show description on click.
I hope you like that, it actually took a lot of time (чяднт?):

```
   CLOCK: [2014-12-08 Mon 21:41]--[2014-12-08 Mon 23:13] =>  1:32 
   CLOCK: [2014-12-08 Mon 16:19]--[2014-12-08 Mon 21:41] =>  5:22
   CLOCK: [2014-12-08 Mon 10:05]--[2014-12-08 Mon 12:04] =>  1:59
   CLOCK: [2014-12-07 Sun 23:20]--[2014-12-08 Mon 01:04] =>  1:44
   CLOCK: [2014-12-07 Sun 18:41]--[2014-12-07 Sun 23:02] =>  4:21
   CLOCK: [2014-12-07 Sun 07:26]--[2014-12-07 Sun 08:14] =>  0:48
   CLOCK: [2014-12-06 Sat 20:41]--[2014-12-07 Sun 01:41] =>  5:00
   CLOCK: [2014-12-06 Sat 10:25]--[2014-12-06 Sat 12:14] =>  1:49
   CLOCK: [2014-12-05 Fri 22:26]--[2014-12-05 Fri 23:34] =>  1:08
   CLOCK: [2014-12-05 Fri 08:26]--[2014-12-05 Fri 08:55] =>  0:29
   CLOCK: [2014-12-03 Wed 17:28]--[2014-12-03 Wed 23:22] =>  5:54
   CLOCK: [2014-12-03 Wed 11:22]--[2014-12-03 Wed 14:52] =>  3:30
   CLOCK: [2014-12-03 Wed 05:15]--[2014-12-03 Wed 10:45] =>  5:30
   CLOCK: [2014-12-02 Tue 22:02]--[2014-12-02 Tue 23:42] =>  1:40
   CLOCK: [2014-12-02 Tue 14:54]--[2014-12-02 Tue 21:00] =>  6:06
   CLOCK: [2014-12-02 Tue 11:34]--[2014-12-02 Tue 12:34] =>  1:00
   CLOCK: [2014-12-02 Tue 05:28]--[2014-12-02 Tue 09:13] =>  3:45
   CLOCK: [2014-12-01 Mon 17:48]--[2014-12-01 Mon 22:56] =>  5:08
   CLOCK: [2014-12-01 Mon 17:08]--[2014-12-01 Mon 17:18] =>  0:10
   CLOCK: [2014-12-01 Mon 05:10]--[2014-12-01 Mon 07:32] =>  2:22
   CLOCK: [2014-11-30 Sun 05:03]--[2014-11-30 Sun 08:20] =>  3:17
   CLOCK: [2014-11-25 Tue 00:50]--[2014-11-25 Tue 02:04] =>  1:14
   CLOCK: [2014-11-24 Mon 23:07]--[2014-11-25 Tue 00:08] =>  1:01
   CLOCK: [2014-11-24 Mon 20:37]--[2014-11-24 Mon 22:52] =>  2:15
   CLOCK: [2014-11-24 Mon 20:10]--[2014-11-24 Mon 20:26] =>  0:16
   CLOCK: [2014-11-24 Mon 18:43]--[2014-11-24 Mon 18:50] =>  0:07
   CLOCK: [2014-11-24 Mon 16:12]--[2014-11-24 Mon 16:38] =>  0:26
   CLOCK: [2014-11-24 Mon 11:36]--[2014-11-24 Mon 12:22] =>  0:46
```

### CODE REVIEW

**Comment by volhovm** [comment]
Everything is really nice, i think. Do not have an ability to test the app on a tablet, but landscape on my phones is normal (portrait is better, of course). I have it on Y.Store -- "MWeather" (it was last year`s task, and I thought...).

**Comment by thevery** [comment]
> Not obvious: forecast items (those with red circles) show description on click.

Obvious enough. Swipe is less obvious actually.

**Comment by thevery** [comment]
- cities should be spinner/navigation drawer rather than separate button
- expand should be closed on expanding another section

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,6 @@
-.DS_Store
-local.properties
-
-*.iml
-
 .gradle
-.idea
```
Comment (thevery) [inline .gitignore:7]: *.iml should be ignored as well

File: app/src/main/res/layout-land/forecast_item.xml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            android:layout_width="190dp"
+            android:layout_height="wrap_content"
+            android:padding="@dimen/android_spaces"
+            android:gravity="center"
+            android:textIsSelectable="true">
+        <TextView
+                android:layout_width="50sp"
+                android:layout_height="50sp"
+                android:padding="@dimen/android_spaces"
+                android:text="27 Sep"
```
Comment (thevery) [inline app/src/main/res/layout-land/forecast_item.xml:21]: tools:text

File: app/src/main/scala/me/volhovm/mweather/WeatherActivity.scala [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+              }
+            }
+          })
+          weatherAddDialog.setButton(DialogInterface.BUTTON_NEUTRAL, "Autodetect", new OnClickListener {
+            override def onClick(p1: DialogInterface, p2: Int): Unit = {
+              val location: Location = mLocationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER)
+              if (location != null) {
+                import me.volhovm.mweather.WeatherLoadService._
+                val intent: Intent = new Intent(Intent.ACTION_SYNC, null, WeatherActivity.this, classOf[WeatherLoadService])
+                Toast.makeText(WeatherActivity.this, "Refreshing", Toast.LENGTH_SHORT).show()
```
Comment (thevery) [inline app/src/main/scala/me/volhovm/mweather/WeatherActivity.scala:199]: `strings.xml`!

**Comment by thevery** [comment]
- add city EditText is not single line and doesn't check input
- design is pretty nice, you should you proper pull-to-refresh with it
- too much toasts on first run

**Comment by thevery** [comment]
Best app so far (despite of scala and 9M+ size)
22

**Pull request closed without merging**